### ISSUE TITLE
No Coverage report available

### ISSUE THREAD

**Issue by IPO-srh-sloan** [rfennell/vNextBuild#240]
### VSTS Extension you are using
Pester
### Where are you running it?
VSTS

### Version of Extension/Task
current public - 6.*

### Expected behaviour and actual behaviour
Now #209  is fixed (thanks) and I can see coverage in the console for my .psm1 files I want to display the report in the VSTS interface but there is no html report generated.

I can see from the build log that coverage ran:
```
Tests Passed: 39, 
Failed: 0, 
Skipped: 0, 
Pending: 0, 
Inconclusive: 0 
Code coverage report:
Covered 21.20 % of 500 analyzed Commands in 13 Files.
Missed commands:
File                                                    Function               
----                                                    --------               
builds\common\Push-ToArtifactory.ps1                                                 
builds\common\VSTSPowershellUtils-Build.psm1            Read-MavenVersions 
```
And coverage-pester.xml gets generated.

Then when the publish code coverage task runs I can see this in the build logs:
```
2018-01-11T17:15:53.0706319Z Reading code coverage summary from 'F:\vstsAgents\pr\int-ss\_work\1\s\Coverage-Pester.XML'
2018-01-11T17:15:53.4596319Z ##[section]Async Command Start: Publish code coverage
2018-01-11T17:15:53.4596319Z Publishing coverage summary data to TFS server.
2018-01-11T17:15:53.4596319Z  Instruction- 106 of 500 covered.
2018-01-11T17:15:53.4596319Z  Line- 82 of 419 covered.
2018-01-11T17:15:53.4596319Z  Method- 11 of 43 covered.
2018-01-11T17:15:53.4596319Z  Class- 2 of 13 covered.
2018-01-11T17:15:53.9716319Z Modifying Cobertura Index file
2018-01-11T17:15:53.9716319Z Publishing code coverage files to TFS server.
2018-01-11T17:15:54.4856319Z Uploading 1 files
2018-01-11T17:15:54.7376319Z File upload succeed.
2018-01-11T17:15:54.7386319Z Published 'F:\vstsAgents\pr\int-ss\_work\_temp\Code Coverage Report_2236' as artifact 'Code Coverage Report_2236'
```
But when I click on the code coverage tab I see this message
`Code coverage data cannot be displayed. Please verify that a valid directory containing code coverage files, including an index.html was used to publish data.`

I don't even see the summary data. I don't know if this is an issue with pester itself, with the plugin or perhaps with vsts displaying jacoco reports...

### Steps to reproduce the problem
Create a VSTS build with a pester task:
![image](https://user-images.githubusercontent.com/13536053/34837437-b8bd9684-f6f2-11e7-86e6-2952161e76cb.png)

And then publish code coverage:
![image](https://user-images.githubusercontent.com/13536053/34837383-8ea0a27e-f6f2-11e7-900d-f08f133c128d.png)
Then run a build and click on the code coverage tab.

**Comment by rfennell** [comment]
I have checked our validation build and see the following

![image](https://user-images.githubusercontent.com/324718/34842386-400aa46e-f703-11e7-84d9-2ab409958617.png)

So I see the code coverage on the summary page, but get the same error on the code coverage tab as you do. 

I suspect we just don't upload the result for the code coverage tab at all - I will look into it

**Comment by ChrisLGardner** [comment]
Sadly that's the only code coverage output that Pester currently outputs. It looks like they're possibly working on some improvements to that, mostly to support other CI tools like CodeCov.io but hopefully that means VSTS can get more data out of the code coverage report.

For progress and discussion on this, check out the Pester issue here: https://github.com/pester/Pester/issues/212

**Comment by rfennell** [comment]
I have marked this is a known issue for now then

@IPO-srh-sloan  do you want this left open? There is no timescale for it to be addressed

**Comment by decapent** [comment]
With the same config as above, I seem to get the following Warning outputted to the console

![image](https://user-images.githubusercontent.com/6863827/34957701-b2154112-f9fc-11e7-8d8b-e7117ea3821d.png)

**Comment by rfennell** [comment]
@decapent as my note above shows (and @ChrisLGardner  explains the reasons why) you will only see the coverage pie charts. There is not HTML report available at present. Is that the issue you meant? if so there is no immediate solution.

Or do you means that you see nothing, this will occur unless you are using Pester 4.0.4 as the message states. This means either

- you have a suitable version of pester installed on the build agent
- you set the properties on the task to used the 4.0.8 version of pester held within the task

**Comment by decapent** [comment]
Indeed, I'm not seeing anything, not even the bar charts. I am using the task's supplied config of using Pester 4.0.8. 

So it is my understanding to initialize my agent with the 4.0.4 version of Pester and configure the task Pester module override ?

**Comment by ChrisLGardner** [comment]
If you look under Advanced on the task you can force the task to use a version shipped with it and selecting 4.0.8 should allow it to use the code coverage functionality.

![image](https://user-images.githubusercontent.com/16837142/34961919-52590410-fa39-11e7-9aaa-67dfe81eacf9.png)

**Comment by decapent** [comment]
Yes this worked perfectly! Thanks for answering quickly <3

I knew i wasn't far off from the solution :)

**Comment by IPO-srh-sloan** [comment]
@rfennell Thanks, might as well close it then as a known issue. TBH I hadn't seen the pie charts on the summary page so at least we get some info :) Would be worth revisiting once pester supports html output.

**Comment by rfennell** [comment]
@IPO-srh-sloan yes, I think we will leave this one as an open known issue until the feature we need in Pester appears

**Comment by rfennell** [comment]
I potential alternative solution has been suggested via the extensions Q&A page on the market place

> Would you consider adding option for exporting report results with ReportUnit? I've found the beautifully formatted report on test results very nice to use, and would be a useful optional export format from the test xml that is generated. 
> 
> If not, a custom powershell script probably could run and do the same thing, but might be an option you and others would benefit from. Github link: https://github.com/reportunit/reportunit 
> 
> I use command similar to Start-Process "C:\packages\ReportUnit.1.5.0-beta1\tools\ReportUnit.exe" -ArgumentList '"C:\temp\.pester" "C:\temp\.pester"' and the result from the folder contents is a beautiful interactive report.

Seems a viable suggestion to investigate
